snooty/rstspec.toml

```diff
@@ -641,6 +641,53 @@ example = """.. card-group::
    ${0: Card content}
 """
 
+[directive."landing:product"]
+content_type = "block"
+help = """Product in a product-list. Each product links to the specified URI, and can include content, an icon, as well as a call-to-action string"""
+options.icon = ["path", "uri"]
+options.icon-alt = "string"
+example = """.. product::
+   ${4::icon: /path-to-icon.svg
+   :icon-alt: Icon's alt text
+
+   ${0:Product content.}
+"""
+
+[directive."landing:products"]
+content_type = "block"
+help = """Container to hold a group of products"""
+example = """.. landing:products::
+   ${0: Product content}
+"""
+
+[directive."landing:client-libraries"]
```

Comment:
does this need an argument? i see that we are passing a heading [here](https://github.com/schmalliso/docs-landing/blob/3099d7975ecfa8f1f018034b35ba7b8a00af5416/source/index.txt#L72)

this is getting parsed to:

```
"children": [
	{
		"type": "paragraph",
		"position": {
			"start": {
				"line": {
					"$numberInt": "71"
				}
			}
		},
		"children": [
			{
				"type": "text",
				"position": {
					"start": {
						"line": {
							"$numberInt": "71"
						}
					}
				},
				"value": "Client Libraries\n:cta: View all\n:url: "
			},
			{
				"type": "reference",
				"position": {
					"start": {
						"line": {
							"$numberInt": "71"
						}
					}
				},
				"children": [
					{
						"type": "text",
						"position": {
							"start": {
								"line": {
									"$numberInt": "71"
								}
							}
						},
						"value": "https://www.mongodb.com/docs/drivers/"
					}
				],
				"refuri": "https://www.mongodb.com/docs/drivers/"
			}
		]
	}
]
```

we would want this separately parsed as a separate heading from the content block, and the `View all` is being treated separately from the reference to drivers

Reply:
@seungpark yes that is correct, but those are just placeholders as this is separate from `DOP-3988`. I need to make those changes for `DOP-3989` as that is the ticket for the client library work. I needed to put something in the as a placeholder to allow the parser to work locally so I could build and test out the UI.